frontend/javascripts/admin/dataset/dataset_upload_view.js

```diff
@@ -446,6 +506,128 @@ class DatasetUploadView extends React.PureComponent<PropsWithFormAndRouter, Stat
   }
 }
 
+const baseStyle = {
+  flex: 1,
+  display: "flex",
+  flexDirection: "column",
+  alignItems: "center",
+  padding: "20px",
+  borderWidth: 2,
+  borderRadius: 2,
+  borderColor: "#eeeeee",
+  borderStyle: "dashed",
+  backgroundColor: "#fafafa",
+  color: "rgba(0, 0, 0, 0.85)",
+  fontSize: 16,
+  outline: "none",
+  transition: "border .24s ease-in-out",
+  cursor: "pointer",
+};
+
+const activeStyle = {
+  borderColor: "#2196f3",
+};
+
+const acceptStyle = {
+  borderColor: "#00e676",
+};
+
+const rejectStyle = {
+  borderColor: "#ff1744",
+};
+
+function FileUploadArea({ fileList, onChange }) {
+  const onDropAccepted = acceptedFiles => {
+    onChange(_.uniqBy(fileList.concat(acceptedFiles), file => file.path));
+  };
+  const removeFile = file => {
+    onChange(_.without(fileList, file));
+  };
+  const { getRootProps, getInputProps, isDragActive, isDragAccept, isDragReject } = useDropzone({
+    onDropAccepted,
+  });
+  const acceptedFiles = fileList;
+
+  const style = useMemo(
+    () => ({
+      ...baseStyle,
+      ...(isDragActive ? activeStyle : {}),
+      ...(isDragAccept ? acceptStyle : {}),
+      ...(isDragReject ? rejectStyle : {}),
+    }),
+    [isDragActive, isDragReject, isDragAccept],
+  );
+
+  const files = acceptedFiles.map(file => <li key={file.path}>{file.path}</li>);
+
+  const showSmallFileList = files.length > 10;
+  const list = (
+    <List
+      itemLayout="horizontal"
+      dataSource={acceptedFiles}
+      size={showSmallFileList ? "small" : "default"}
+      renderItem={item => (
+        <List.Item
+          actions={[
+            <a key="list-loadmore-edit" onClick={() => removeFile(item)}>
```

Comment:
Shouldn't these List.Items all have a different key?

Reply:
Hmm, doesn't seem to be necessary. There is no warning in the console. I assume that the `<List>` component adds the key property after calling `renderItem` :thinking: